axis.go

```diff
@@ -497,6 +498,158 @@ func (utt UnixTimeTicks) Ticks(min, max float64) []Tick {
+// adjust the number of ticks according to the specified Width.  If
+// not specified, Width defaults to 4 inches.
+type AutoUnixTimeTicks struct {
+	// Width is the width of the underlying graph, used to calculate
+	// the number of ticks that can fit properly with their time
+	// shown.
+	Width vg.Length
+}
+
+var _ Ticker = AutoUnixTimeTicks{}
+
+// Inspired by https://github.com/d3/d3-scale/blob/master/src/time.js
+var tickIntervals = []tickRule{
+	{time.Millisecond,
+		"15:04:05.000", "15:04:05", ".000", time.Millisecond},
+	{200 * time.Millisecond,
+		"15:04:05.000", "15:04:05", ".000", 200 * time.Millisecond},
+	{500 * time.Millisecond,
+		"15:04:05.000", "15:04:05", ".000", 500 * time.Millisecond},
+	{time.Second,
+		"15:04:05", "15:04", ":05", time.Second},
+	{2 * time.Second,
+		"15:04:05", "15:04", ":05", 2 * time.Second},
+	{5 * time.Second,
+		"15:04:05", "15:04", ":05", 5 * time.Second},
+	{15 * time.Second,
+		"Jan 02, 15:04", "Jan 02", "15:04", 15 * time.Second},
+	{30 * time.Second,
+		"15:04:05", "15:04", ":05", 30 * time.Second},
+	{time.Minute,
+		"15:04:05", "15:04", ":05", time.Minute},
+	{2 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 2 * time.Minute},
+	{5 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 5 * time.Minute},
+	{15 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 15 * time.Minute},
+	{30 * time.Minute,
+		"Jan 02, 3:04pm", "Jan 02", "3:04pm", 30 * time.Minute},
+	{time.Hour,
+		"Jan 2, 3pm", "Jan 2", "3pm", time.Hour},
+	{3 * time.Hour,
+		"Jan 2, 3pm", "Jan 2", "3pm", 3 * time.Hour},
+	{6 * time.Hour,
+		"Jan 2, 3pm", "Jan 2", "3pm", 6 * time.Hour},
+	{12 * time.Hour,
+		"Jan 2", "Jan 2", "3pm", 12 * time.Hour},
+	{24 * time.Hour,
+		"Jan 2", "Jan", "2", 24 * time.Hour},
+	{48 * time.Hour,
+		"Jan 2", "Jan", "2", 48 * time.Hour},
+	{7 * 24 * time.Hour,
+		"Jan 2", "Jan", "2", 7 * 24 * time.Hour},
+	{aMonth, // 1 month
+		"Jan 2006", "2006", "Jan", aMonth},
+	{3 * aMonth, // 3 months
+		"Jan 2006", "2006", "Jan", 3 * aMonth},
+	{6 * aMonth, // 6 months
+		"Jan 2006", "2006", "Jan", 6 * aMonth},
+	{12 * aMonth, // 1 year
+		"2006", "", "2006", 12 * aMonth},
+	{2 * 12 * aMonth, // 2 years
+		"2006", "", "2006", 2 * 12 * aMonth},
+	{5 * 12 * aMonth, // 5 years
+		"2006", "", "2006", 5 * 12 * aMonth},
+	{10 * 12 * aMonth, // 10 years
+		"2006", "", "2006", 10 * 12 * aMonth},
+}
+
+var aMonth = 31 * 24 * time.Hour
+
+type tickRule struct {
+	DurationPerInch time.Duration // use this rule for a maximum Duration per inch
+	LongFormat      string        // longer format
+	WatchFormat     string        // show long format when WatchFormat changes between ticks
+	ShortFormat     string        // incremental format, shorter
+	TimeWindow      time.Duration // interval for ticks
+}
+
+// Ticks implements plot.Ticker.
+func (t AutoUnixTimeTicks) Ticks(min, max float64) []Tick {
+	width := t.Width
+	if width == 0 {
+		width = 4 * vg.Inch
+	}
+
+	minT := time.Unix(int64(min), 0).UTC()
```

Comment:
It's there to distinguish from the function parameters `min` and `max`, shortly indicating they're `time.Time` instead of a `float64`.  Would you prefer some other names ?